central/networkgraph/entity/datastore/internal/store/postgres/full_store.go

```diff
@@ -0,0 +1,85 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/stackrox/rox/central/metrics"
+	"github.com/stackrox/rox/central/networkgraph/entity/datastore/internal/store"
+	"github.com/stackrox/rox/pkg/env"
+	ops "github.com/stackrox/rox/pkg/metrics"
+	"github.com/stackrox/rox/pkg/postgres"
+	pkgSchema "github.com/stackrox/rox/pkg/postgres/schema"
+)
+
+const (
+	networkFlowsTable    = pkgSchema.NetworkFlowsTableName
+	networkEntitiesTable = pkgSchema.NetworkEntitiesTableName
+
+	pruneOrphanExternalNetworkEntitiesSrcStmt = `DELETE FROM %s entity WHERE NOT EXISTS
+		(SELECT 1 FROM %s flow WHERE flow.Props_SrcEntity_Type = 4
+		AND flow.Props_SrcEntity_Id = entity.Info_Id
+		AND entity.Info_ExternalSource_Discovered = true);`
```

Comment:
So these are going against the `network_flow` main partition insead of the individual partitions of the cluster.  That could have some performance impacts as it will need to run against all partitions and not just the specific one for the cluster.  Without a good scale environment it is hard to quantify how big an issue that will be or not.

Reply:
Unfortunately, I don't see a way to achieve pruning while respecting the netflows partitioning. Indeed, we are searching for entities which are not used by "any" flow, so we can't rely on the result from only one partition.

We have plans to enhance this for 4.7 (https://issues.redhat.com/browse/ROX-26854) through pagination. In the meantime, I propose to merge this one to have at least a pruning method available.